Intermittent object test failure for bucket

A recent build failed with this error
https://jenkins.rook.io/blue/organizations/jenkins/rook%2Frook/detail/PR-379/3/pipeline/20
```
--- FAIL: TestListBuckets (0.23s)
251 		assertions.go:109: 
                        
	Location:	object_test.go:581
252 			
	Error:		Not equal: []string{"metadata", "get", "--cluster=default", "--conf=/tmp/rgw/tmp/default.config", "--keyring=/tmp/rgw/tmp/keyring", "bucket:foo"} != []string{"metadata", "get", "--cluster=default", "--conf=/tmp/rgw/tmp/default.config", "--keyring=/tmp/rgw/tmp/keyring", "bucket:bar"}
253 			
254 		assertions.go:109: 
                        
	Location:	object_test.go:584
255 			
	Error:		Not equal: []string{"metadata", "get", "--cluster=default", "--conf=/tmp/rgw/tmp/default.config", "--keyring=/tmp/rgw/tmp/keyring", "bucket:bar"} != []string{"metadata", "get", "--cluster=default", "--conf=/tmp/rgw/tmp/default.config", "--keyring=/tmp/rgw/tmp/keyring", "bucket:foo"}
256 			
257 		assertions.go:109: 
                        
	Location:	object_test.go:591
258 			
	Error:		Not equal: "[{\"name\":\"foo\",\"owner\":\"bob\",\"createdAt\":\"2016-08-05T16:23:34.343343Z\",\"size\":4,\"numberOfObjects\":2},{\"name\":\"bar\",\"owner\":\"bill\",\"createdAt\":\"2016-08-05T18:31:22.445343Z\",\"size\":5,\"numberOfObjects\":4}]" != "[{\"name\":\"bar\",\"owner\":\"bob\",\"createdAt\":\"2016-08-05T16:23:34.343343Z\",\"size\":5,\"numberOfObjects\":4},{\"name\":\"foo\",\"owner\":\"bill\",\"createdAt\":\"2016-08-05T18:31:22.445343Z\",\"size\":4,\"numberOfObjects\":2}]"
259 			
```

--- a/pkg/api/filesystem_test.go
+++ b/pkg/api/filesystem_test.go
@@ -21,6 +21,7 @@ import (
 	"io/ioutil"
 	"net/http"
 	"net/http/httptest"
+	"os"
 	"strings"
 	"testing"
 
@@ -83,6 +84,7 @@ func TestCreateFileSystemHandler(t *testing.T) {
 	etcdClient := util.NewMockEtcdClient()
 	context := &clusterd.Context{EtcdClient: etcdClient}
 	test.CreateClusterInfo(etcdClient, []string{"a"})
+	defer os.RemoveAll("mon0")
 
 	req, err := http.NewRequest("POST", "http://10.0.0.100/filesystem", strings.NewReader(`{"name": "myfs1", "poolName": "myfs1-pool"}`))
 	if err != nil {

--- a/pkg/api/object.go
+++ b/pkg/api/object.go
@@ -16,9 +16,9 @@ limitations under the License.
 package api
 
 import (
-	"net/http"
-
 	"encoding/json"
+	"net/http"
+	"sort"
 
 	"github.com/gorilla/mux"
 	"github.com/rook/rook/pkg/cephmgr/rgw"
@@ -215,7 +215,10 @@ func (h *Handler) Listbuckets(w http.ResponseWriter, r *http.Request) {
 		return
 	}
 
-	FormatJsonResponse(w, buckets)
+	sortedBuckets := model.ObjectBuckets(buckets)
+	sort.Sort(sortedBuckets)
+
+	FormatJsonResponse(w, sortedBuckets)
 }
 
 func handleConnectionLookupFailure(err error, action string, w http.ResponseWriter) {

--- a/pkg/api/object_test.go
+++ b/pkg/api/object_test.go
@@ -588,5 +588,5 @@ func TestListBuckets(t *testing.T) {
 		return "", fmt.Errorf("Shouldn't return more than 3 times")
 	})
 	assert.Equal(t, http.StatusOK, w.Code)
-	assert.Equal(t, `[{"name":"foo","owner":"bob","createdAt":"2016-08-05T16:23:34.343343Z","size":4,"numberOfObjects":2},{"name":"bar","owner":"bill","createdAt":"2016-08-05T18:31:22.445343Z","size":5,"numberOfObjects":4}]`, w.Body.String())
+	assert.Equal(t, `[{"name":"bar","owner":"bill","createdAt":"2016-08-05T18:31:22.445343Z","size":5,"numberOfObjects":4},{"name":"foo","owner":"bob","createdAt":"2016-08-05T16:23:34.343343Z","size":4,"numberOfObjects":2}]`, w.Body.String())
 }

--- a/pkg/model/object.go
+++ b/pkg/model/object.go
@@ -41,3 +41,17 @@ type ObjectBucket struct {
 	CreatedAt time.Time `json:"createdAt"`
 	ObjectBucketStats
 }
+
+type ObjectBuckets []ObjectBucket
+
+func (slice ObjectBuckets) Len() int {
+	return len(slice)
+}
+
+func (slice ObjectBuckets) Less(i, j int) bool {
+	return slice[i].Name < slice[j].Name
+}
+
+func (slice ObjectBuckets) Swap(i, j int) {
+	slice[i], slice[j] = slice[j], slice[i]
+}

